dijet/selection/jet.py

```diff
@@ -0,0 +1,60 @@
+# coding: utf-8
+
+from typing import Tuple
+from columnflow.util import maybe_import
+from columnflow.columnar_util import set_ak_column
+from columnflow.selection import Selector, SelectionResult, selector
+
+ak = maybe_import("awkward")
+
+
+def masked_sorted_indices(mask: ak.Array, sort_var: ak.Array, ascending: bool = False) -> ak.Array:
+    """
+    Helper function to obtain the correct indices of an object mask
+    """
+    indices = ak.argsort(sort_var, axis=-1, ascending=ascending)
+    return indices[mask[indices]]
+
+
+@selector(
+    uses={"Jet.pt", "Jet.eta", "Jet.phi", "Jet.jetId"},
+    produces={"cutflow.n_jet"},
+    exposed=True,
+)
+def jet(
+    self: Selector,
+    events: ak.Array,
+    **kwargs,
+) -> Tuple[ak.Array, SelectionResult]:
+    # DiJet jet selection
+    # - require ...
+
+    # assign local index to all Jets - stored after masks for matching
+    # TODO: Drop for dijet ?
+    events = set_ak_column(events, "local_index", ak.local_index(events.Jet))
```

Comment:
The `local_index` refers to jets, but the column name does not indicate this. You could define a new subfield of `Jet` to store the index if you need it in a different selector:
```suggestion
    events = set_ak_column(events, "Jet.local_index", ak.local_index(events.Jet))
```

Also, unless you also add the column to the `produces`, it will not be stored.